addons/l10n_be_hr_payroll_fleet/models/fleet.py

```diff
@@ -45,13 +45,17 @@ def _compute_total_cost(self):
                 elif contract.cost_frequency == "yearly":
                     car.total_cost += contract.cost_generated / 12.0
 
-    def _get_co2_fee(self, co2):
-        return max((((co2 * 9.0) - 600.0) * 1.2488) / 12.0, 0.0)
-
-    @api.depends('co2')
+    def _get_co2_fee(self, co2, fuel_type):
+        fuel_coefficient = {'diesel': 600, 'gasoline': 768, 'lpg': 990, 'electric': 0, 'hybrid': 0}
+        co2_fee = 26.47
+        if fuel_type and fuel_type not in ['electric', 'hybrid']:
+            co2_fee = (((co2 * 9.0) - fuel_coefficient.get(fuel_type)) * 1.2488) / 12.0
```

Comment:
Here you use `fuel_coefficient.get(fuel_type)` which in case fuel_type is not found in fuel_coefficient will return None
As None cannot be multiplied by a number in python an error will be raised, but the error message will not contain much informations
I advise to use either `fuel_coefficient[fuel_type` which will raise a more informative error or `fuel_coefficient.get(fuel_type, 0)` which will not raise any error but will consider any fuel_type not found  as if it was electric or hybrid